src/monster.cpp

```diff
@@ -1216,6 +1216,21 @@ int monster::hit_roll() const {
     return dice(type->melee_skill, 10);
 }
 
+bool monster::has_grab_break_tec() const
+{
+    return false;
+}
+
+bool monster::stability_check( monster *z, int strength) const
```

Comment:
You should only pass the difficulty to `stability_check`. Caller should set the difficulty.

You'd call it like that:
```
if( !target->stability_check( dice( z->type->melee_dice, z->type->melee_damage ) - rng( 0, 10 ) ) {
    // Get pulled
}
```
And in the stability check you'd do the defender's part of the roll (the part without `z->`) and compare it against the difficulty.